packages/app-dashboard/src/components/withdraw/WalletConnect/WalletConnectUtil.ts

```
@@ -0,0 +1,290 @@
+ */
+export async function createWalletConnectClient(
+  setClient?: (client: IWalletKit) => void,
+  forceReset = false,
+): Promise<IWalletKit> {
+  // If we're forcing a reset, clean up the existing client
+  if (forceReset && walletKitClient) {
+    console.log('Force resetting WalletKit client');
+    await resetWalletConnectClient();
+  }
+
+  // If already initialized, return existing instance
+  if (walletKitClient) {
+    if (setClient) {
+      setClient(walletKitClient);
+    }
+    return walletKitClient;
+  }
+
+  // Prevent concurrent initialization
+  if (isInitializing) {
+    console.log('WalletKit initialization in progress, waiting...');
+    // Wait for initialization to complete
+    return new Promise<IWalletKit>((resolve) => {
+      const checkInterval = setInterval(() => {
+        if (walletKitClient && !isInitializing) {
+          clearInterval(checkInterval);
+          if (setClient) {
+            setClient(walletKitClient);
+          }
+          resolve(walletKitClient);
+        }
+      }, 100);
+    });
+  }
+
+  try {
+    isInitializing = true;
+    console.log('Initializing new WalletKit instance');
+
+    // Create a shared Core instance
+    const core = new Core({
+      projectId: VITE_WALLETCONNECT_PROJECT_ID,
+    });
+
+    // Initialize WalletKit with the required parameters - according to documentation
+    walletKitClient = await WalletKit.init({
+      core, // Pass the shared core instance
+      metadata: {
+        name: 'Vincent App',
+        description: 'Vincent App using PKP with WalletKit',
+        url: window.location.origin,
+        icons: ['https://lit.network/favicon.ico'], // TODO: Add a logo
+      },
+    });
+
+    // Store the client in the store if a setter was provided
+    if (setClient) {
+      setClient(walletKitClient);
+    }
+
+    return walletKitClient;
+  } catch (error) {
+    console.error('Error initializing WalletKit:', error);
+    throw error;
+  } finally {
```

Don't just log and re-throw. If the error is not handled just let it bubble up without catching or the logs will pollute with a bunch of repeated logs real quick